docs/devsite-help/protobuf.md

```diff
@@ -0,0 +1,142 @@
+# Protocol buffers in Google Cloud Libraries for .NET
+
+[Protocol buffers](https://developers.google.com/protocol-buffers) - also known as *protobuf* for short - are Google's language-neutral, platform-neutral, extensible mechanism for serializing structured data.
+
+Almost all the Google Cloud Libraries are generated from API
+definitions using protobuf, in the
+[GitHub googleapis/googleapis repository](https://github.com/googleapis/googleapis).
+
+Most protobuf messages in the APIs are simply generated with `protoc`
+(the Protocol Buffer Compiler) using the vanilla C# plugin. However,
+the generated partial classes are augmented in some libraries:
+
+- [Pagination](page-streaming.md) requires request/response types
+  to implement specific interfaces from GAX.
+- [Resource names](resource-names.md) add resource-name-typed properties
+  as strongly-typed "views" over string fields declared in the messages.
+- In some libraries, classes are augmented to provide simpler ways of
+  creating instances, such as via the `Filter` class in Datastore.
+
+For the most part, protobuf objects can be treated like normal mutable
+data transfer objects which happen to have well-defined binary and JSON
+formats. The sections below provide more information in areas which have
+previously caused some confusion.
+
+# Repeated fields and maps
+
+The generated C# code for protobuf messages makes simple properties
+read/write, but repeated fields and map fields are read-only. That
+doesn't stop you from populating them, though: it just means you
+can't change the property to refer to a *different* list or map.
+
+Typically you'll populate this using a *collection initializer*
+nested within an *object initializer*. As an example, let's look at
+how we might create a `BatchAnnotateImagesRequest` message in the
+Vision API. (This is just an easy-to-understand example; the
```

Comment:
nit:
Maybe find an example where there are no helper methods?

Reply:
Or maybe just use a FooMessage? This is where that FOO API would help a lot.

Reply:
I'd prefer to use a real API so that we can have code that compiles.
This example was basically moved from the previous troubleshooting guide - I think I'd prefer to keep it as-is for now, and we can revisit later.